aesara/tensor/basic.py

```diff
@@ -1254,6 +1255,122 @@ def triu(m, k=0):
     )
 
 
+def tril_indices(
+    n: Union[int, ScalarVariable],
+    k: Union[int, ScalarVariable] = 0,
+    m: Optional[Union[int, ScalarVariable]] = None,
+) -> Tuple[TensorVariable, TensorVariable]:
+    """
+    Return the indices for the lower-triangle of an (n, m) array.
+
+    Parameters
+    ----------
+    n : integer scalar
+        The row dimension of the arrays for which the returned indices will be valid.
+    k : integer scalar, optional
+        Diagonal offset to use when forming the indices. `k = 0` (the default)
+        is the main diagonal, `k < 0` is below it and `k > 0` is above.
+    m : integer scalar, optional
+        The column dimension of the arrays for which the returned arrays will
+        be valid. By default m is taken equal to n.
+
+    Returns
+    -------
+    inds : tuple of TensorVariable's
+        The indices for the triangle. The returned tuple contains two arrays,
+        each with the indices along one dimension of the array.
+    """
+    return tri(n, m, k, dtype=bool).nonzero()
+
+
+def tril_indices_from(
+    a: Union[np.ndarray, matrix],
```

Comment:
are `row` and `col` valid inputs besides `matrix`? I am not sure about typehints for specific shape TensorVariables. CC @brandonwillard

Reply:
Im not sure if it would make sense to get triangular indices for a row or col. It would simply be the element in the `k`'th index.  IMHO I dont think we need to worry about that level of granularity with the type hints.

Reply:
My question is not limited to the row and column matrices (which should be supported as long as numpy also supports them)

The other point is that we now treat TensorVariables with fixed shapes uniquely. A TensorVariables with known `shape=(2,2)` (a matrix with fixed 2×2 shape) for instance is now a different type than a TensorVariable with unknown `shape=(None, None)`, which is what that `matrix` in your type hints defines.

Reply:
added row and col hints.

Reply:
I still think the best you can do is to say it accepts `TensorVariable`.

If I understand correctly, a `TensorVariable` with say `shape=(2, 2)` (or any other `shape=(n|None, m|None))` is not of the same type as a `matrix`, which is just a specific `TensorVariable` with `shape=(None, None)`

Reply:
removed row and row and replaced with TensorVariable.